### PR TITLE
[FEAT] Goal Command API

### PR DESCRIPTION
**Pull request by sueelly**
## #️⃣ Related Issue

Linked Feature backlog: [issue#3](https://github.com/dnd-side-project/dnd-12th-2-backend/issues/3)

## 📝 Purpose of PR

Goal Command API를 구현한다.

## Contents

Controller / repository test는 시간관계상 생략했다.
실제 환경과 가장 유사한 Acceptance test 주 로직-Service layer에 대한 단위테스트만 존재한다.

- [x] Goal Create, Achieve, Delete API
- [x] Passed Acceptance Test
- [x] Add sample data for test

## Checklist

-   [x] Does commit messages follow the convention?
-   [x] Are variable names brief but descriptive?
-   [x] Is the code clean and easy to understand?
-   [x] If a new feature has been added, or a bug fixed, has a test been added to confirm good behavior?
-   [x] Does the test successfully test edge and corner cases?

## 💬 Review Requirements (Optional)

Is there anything you can praise about this PR? Start with praise.
Make a review, leaving kind comments and suggesting changes where needed (to resolve the above).

### CODE REVIEW

File: src/main/java/ac/dnd/dodal/common/model/BaseEntity.java [hunk 1]
```diff
@@ -25,10 +25,10 @@
 public class BaseEntity {
 
     @Column(updatable = false)
-    @CreatedDate
+    @CreationTimestamp
```
Comment (5hseok) [inline src/main/java/ac/dnd/dodal/common/model/BaseEntity.java:28]: 기존 어노테이션에서 hibernate의 어노테이션으로 수정한 이유는 뭘까요?

File: src/main/resources/db/migration/V3__sample_data.sql
Comment (5hseok) [inline src/main/resources/db/migration/V3__sample_data.sql:1]: V3가 이미 존재하니 V4로 바꾸는건 어떠신가요?

File: src/test/java/ac/dnd/dodal/acceptance/goal/AchieveGoalAcceptanceTest.java
Comment (5hseok) [inline src/test/java/ac/dnd/dodal/acceptance/goal/AchieveGoalAcceptanceTest.java:1]: 제 쪽에서는 HeloAcceptanceTest를 제외하면 인수 테스트가 안되던데, 혜주님 로컬에서는 잘 되시나요?

**Review by 5hseok — CHANGES_REQUESTED** [review]

File: src/main/java/ac/dnd/dodal/common/model/BaseEntity.java (hunk 1 above)
Comment (sueelly) [reply src/main/java/ac/dnd/dodal/common/model/BaseEntity.java:28]: 저거 안돌아갑니다!!! 정확히는 값이 삽입이 안돼요
근데 왜 안돌아가는지까지는 모르겠습니다.
똑같은 조건에서 저거만 안되길래 우선 수정하고, 이후에 찾아보자 하고 넘겼습니다.

File: src/test/java/ac/dnd/dodal/acceptance/goal/AchieveGoalAcceptanceTest.java
Comment (sueelly) [reply src/test/java/ac/dnd/dodal/acceptance/goal/AchieveGoalAcceptanceTest.java:1]: 제 로컬에서는 잘 됩니다. 현석님 mockMVC 테스트를 한번 주석처리해보시겠어요?
+ 에러 로그 올려주시면 좋을 것 같습니다. Caused by 중심으로

Comment (sueelly) [reply src/test/java/ac/dnd/dodal/acceptance/goal/AchieveGoalAcceptanceTest.java:1]: image
